modules/siddhi-extensions/output-mappers/xml-output-mapper/src/main/java/org/wso2/siddhi/extension/output/mapper/xml/XMLOutputMapper.java

```diff
@@ -20,53 +20,132 @@
         this.streamDefinition = streamDefinition;
+        if (payloadTemplateBuilder != null) {
+            //Means, this is a custom mapping scenario; In default case, prefix & suffix is ignored.
+            Option prefixXmlOption = optionHolder.getOrCreateOption(OPTION_PREFIX_XML, null);
+            if (prefixXmlOption != null) {
+                prefixXml = prefixXmlOption.getValue();
+            }
+            Option suffixXmlOption = optionHolder.getOrCreateOption(OPTION_SUFFIX_XML, null);
+            if (suffixXmlOption != null) {
+                suffixXml = suffixXmlOption.getValue();
+            }
+            if (prefixXml != null && suffixXml == null) {
+                log.warn(OPTION_SUFFIX_XML + " option is not given while " + OPTION_PREFIX_XML + " option has been given. " +
+                        "Typically, these options are used together.");
+            } else if (suffixXml != null && prefixXml == null) {
+                log.warn(OPTION_PREFIX_XML + " option is not given while " + OPTION_SUFFIX_XML + " option has been given. " +
+                        "Typically, these options are used together.");
+            }
+        }
+        Option validateXmlOption = optionHolder.getOrCreateOption(OPTION_VALIDATE_XML, null);
+        if (validateXmlOption != null) {
+            xmlValidationEnabled = Boolean.parseBoolean(validateXmlOption.getValue());
+        }
     }
 
     /**
-     * Convert the given Event mapping to XML string
+     * Map and publish the given {@link Event} array
      *
-     * @param event         Event object
-     * @param mappedPayload mapped Payload if any
-     * @return the mapped XML string
+     * @param events                  Event object array
+     * @param outputTransportCallback output transport callback
+     * @param optionHolder            option holder containing static and dynamic options
+     * @param payloadTemplateBuilder  Unmapped payload for reference
      */
     @Override
-    public Object mapEvent(Event event, String mappedPayload) {
-        if (mappedPayload != null) {
-            return mappedPayload;
+    public void mapAndSend(Event[] events, OutputTransportCallback outputTransportCallback,
+                           OptionHolder optionHolder, TemplateBuilder payloadTemplateBuilder)
+            throws ConnectionUnavailableException {
+        if (events.length < 1) {
+            return;
+        }
+        StringBuilder sb = new StringBuilder();
+        if (payloadTemplateBuilder != null) {   //custom mapping
+            if (prefixXml != null) {
+                sb.append(prefixXml);
+            }
+            for (Event event : events) {
+                sb.append(payloadTemplateBuilder.build(event));
+            }
+            if (suffixXml != null) {
+                sb.append(suffixXml);
+            }
+            if (xmlValidationEnabled) {
+                DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
```

Comment:
Thanks for pointing this out. Fixed in https://github.com/wso2/siddhi/pull/268/commits/0c7c705ba49480d7ae399d7fd197b24965d98731